From 65be479e7760db53dd7a42eb70abc568e6ccec85 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Sat, 6 Jul 2013 16:44:35 +0200
Subject: [PATCH 081/127] wined3d: Accelerate READONLY buffer maps
Reply-To: wine-devel <wine-devel@winehq.org>

FIXME: ProcessVertices can write to buffers
---
 dlls/wined3d/buffer.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/wined3d/buffer.c b/dlls/wined3d/buffer.c
index 4447908..57bc169 100644
--- a/dlls/wined3d/buffer.c
+++ b/dlls/wined3d/buffer.c
@@ -1075,9 +1075,9 @@ HRESULT CDECL wined3d_buffer_map(struct wined3d_buffer *buffer, UINT offset, UIN
             buffer->map_mem = wined3d_resource_allocate_sysmem(buffer->resource.size);
             wined3d_cs_emit_swap_mem(device->cs, buffer, buffer->map_mem);
         }
-        else if(!(flags & WINED3D_MAP_NOOVERWRITE))
+        else if(!(flags & (WINED3D_MAP_NOOVERWRITE | WINED3D_MAP_READONLY)))
         {
-            FIXME("waiting for cs.\n");
+            FIXME("waiting for cs, flags 0x%04x.\n", flags);
             device->cs->ops->finish(device->cs);
         }
     }
-- 
1.8.1.5

